internal/subscription/core.go

```diff
@@ -497,3 +501,60 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 	logger.Ctx(ctx).Infow("migration: request completed.", "subscriptionsUpdated", updatedSubCount)
 	return nil
 }
+
+func (c *Core) deleteSubscriptionTopics(ctx context.Context, m *Model) error {
+	if m == nil || m.GetTopic() == "" {
+		return errors.New("Nil model")
+	}
+
+	projectID := m.ExtractedTopicProjectID
+
+	//  delete internal topic
+	internalTopic := m.GetSubscriptionTopic()
```

Comment:
build an array of topics to be deleted and execute in a loop. This is cumbersome to maintain

Reply:
Done